lib/puppet/provider/scheduled_task/win32_taskscheduler.rb

```diff
@@ -141,6 +146,10 @@ def command=(value)
     task.application_name = value
   end
 
+  def comment=(value)
+    task.comment = value
```

Comment:
There's not a specific maximum length according the API, but we can put a limit;
In the Task Scheduler GUI, 260 chars is the maximum input for a comment which seems to represent MAX_PATH. Everything longer will be truncated when you press Apply/OK if you can actually get it in there (you could using the API! but as said it will be truncated if you make a manual change anyway).
Same goes for command/arguments (and probably the others like account)..also 260 max in gui when starting fresh, but if you got a bigger amount in it seems to support adding even more...! No true limit there really.
I propose changing puppet/util/windows/taskscheduler.rb#702 to 260 (for comment) and #408 to 512 (for arguments) for now which should be enough for most cases(?). And all others using 256 as max should be changed to 260 as well (like working dir etc), because I think it's logical to use MAX_PATH (https://msdn.microsoft.com/en-us/library/aa370993%28v=vs.85%29.aspx)
We could add a check on the length of all these parameters and fail if user supplies a longer argument...do you want that? (and kind of workaround for PUP-3768 on that). I did not change anything regarding this, awaiting your comments.